src/language-markdown/print-preprocess.js

```diff
@@ -34,7 +34,13 @@ function transformInlineCode(ast) {
       return node;
     }
 
-    return { ...node, value: node.value.replace(/\s+/g, " ") };
+    return {
+      ...node,
+      value: node.value
+        .replace(/^\s+/, " ")
+        .replace(/\n/g, " ")
+        .replace(/\s+$/, " "),
```

Comment:
Does `/(?:^\s+|\n|\s+$)/g` work? Is it faster?

Reply:
I‘ve updated the logic a bit. Looks like we need to treat ` ` and other whitespace differently, [according to the spec](https://spec.commonmark.org/0.30/#backtick-string).